src/kubernetes_cluster/mod.rs

```diff
@@ -2,3 +2,11 @@
 // SPDX-License-Identifier: MIT
 pub mod proof;
 pub mod spec;
+
+use std::marker::PhantomData;
+
+pub struct Cluster<K, E, R> {
```

Comment:
Could you document why we need this empty Cluster struct and also the three phantom markers?

Reply:
Why the empty struct? We use the struct to hold all the functions. It has not fields so it is empty.

Reply:
Yes, I know that. I am basically asking to document:
(1) why we want to move so many spec/proof fn into the Cluster struct
(2) what is the purpose of the three phantom fields (which I believe are used to suppress some Verus warnings)